smelter-core/src/pipeline/hls/hls_input.rs

```diff
@@ -47,6 +47,18 @@ struct Track {
     index: usize,
     handle: DecoderThreadHandle,
     state: StreamState,
+    stats_sender: HlsInputTrackStatsSender,
+}
+
+impl Track {
+    fn pts_dts_from_packet(&mut self, packet: &Packet) -> (Duration, Option<Duration>) {
+        self.state.pts_dts_from_packet(packet, &self.stats_sender)
+    }
+
+    fn send_stats_on_packet_received(&self, packet: &Packet, packet_pts: Duration) {
+        self.stats_sender
+            .send_on_packet_received(packet, packet_pts, &self.state);
+    }
```

Comment:
I think those helpers are unnecessary, they are doing to little to justify. You can avoid part of this logic by passing stats_sender to in the state creation